testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -447,6 +447,81 @@ public void testTrinoReadsSparkPartitionedTable(StorageFormat storageFormat, int
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS, ICEBERG_REST, ICEBERG_JDBC}, dataProvider = "storageFormats")
+    public void testSparkReadsTrinoNestedPartitionedTable(StorageFormat storageFormat)
+    {
+        String baseTableName = toLowerCase("test_spark_reads_trino_nested_partitioned_table_" + storageFormat + randomNameSuffix());
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + trinoTableName);
+        onTrino().executeQuery(format(
+                "CREATE TABLE %s (_string VARCHAR, _varbinary VARBINARY, _bigint BIGINT, _struct ROW(_field BIGINT))" +
+                        " WITH (partitioning = ARRAY['\"_struct._field\"'], format = '%s')",
+                trinoTableName,
+                storageFormat));
+        onTrino().executeQuery(format(
+                "INSERT INTO %s VALUES" +
+                        " ('a', X'0ff102f0feff', 1001, ROW(1))," +
+                        " ('b', X'0ff102f0fefe', 1002, ROW(2))," +
+                        " ('c', X'0ff102fdfeff', 1003, ROW(3))",
+                trinoTableName));
+
+        Row expectedRow = row("a", new byte[]{15, -15, 2, -16, -2, -1}, 1001, 1);
+        String select = "SELECT _string, _varbinary, _bigint, _struct._field FROM %s WHERE _string = 'a'";
+        assertThat(onTrino().executeQuery(format(select, trinoTableName)))
+                .containsOnly(expectedRow);
+        if (storageFormat == StorageFormat.ORC) {
+            assertThatThrownBy(() -> onSpark().executeQuery(format(select, sparkTableName)))
+                    .hasMessageContaining("java.lang.IndexOutOfBoundsException: Index 3 out of bounds for length 3");
+        }
+        else {
+            if (storageFormat == StorageFormat.PARQUET) {
+                // Spark returns null for nested partitioned field's value
+                expectedRow = row("a", new byte[]{15, -15, 2, -16, -2, -1}, 1001, null);
+            }
+            assertThat(onSpark().executeQuery(format(select, sparkTableName)))
+                    .containsOnly(expectedRow);
+        }
+        onTrino().executeQuery("DROP TABLE " + trinoTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS, ICEBERG_REST, ICEBERG_JDBC}, dataProvider = "storageFormatsWithSpecVersion")
+    public void testTrinoReadsSparkNestedPartitionedTable(StorageFormat storageFormat, int specVersion)
```

Comment:
Does the specVersion matter?
add a comment why.

Reply:
`specVersion` does not matter. removed.